src/test/util/setup_common.cpp

```diff
@@ -187,8 +188,9 @@ TestingSetup::TestingSetup(const std::string& chainName, const std::vector<const
         throw std::runtime_error(strprintf("ActivateBestChain failed. (%s)", state.ToString()));
     }
 
+    m_node.addrman = std::make_unique<CAddrMan>();
```

Comment:
8f3d35a83affc7061e6bf9b9e6be6e9861c84632:

Any reason to implicitly deconstruct this last, as opposed to in the same order that `init` deconstructs it? This will leave a "dangling" addrman in memory after the datadir (and addrman.dat) is cleared.

Reply:
Just to be clear, are you suggesting a `~TestingSetup()` dtor that destructs the components in reverse order, like in `Shutdown()`? That seems reasonable to me.

Reply:
No, just add `m_node.addrman.reset()` to where the other node members are reset.

Reply:
Done. Seems odd to me that the components are constructed in the `TestingSetup()` ctor, but destructed in the `ChainTestingSetup()` destructor, but I've followed the existing pattern.